src/gc/gc.d

```diff
@@ -2803,19 +2799,36 @@ struct Pool
         }
     }
 
-    void clrBitsSmallSweep(size_t dataIndex, GCBits.wordtype toClear) nothrow
+    void freePageBits(size_t pagenum, in ref PAGEBITS toFree) nothrow
     {
-        immutable toKeep = ~toClear;
+        import core.internal.traits : staticIota;
+
+        immutable beg = pagenum * (PAGESIZE / 16 / GCBits.BITS_PER_WORD);
+
+        foreach (i; staticIota!(0, PAGEBITS.length))
```

Comment:
staticIota should unroll the loops to 8 or 4 operations for 32 or 64 bit, respectively, operating on a single cache line. I guess that's difficult to beat.